build_runner/test/build_script_generate/build_script_generate_test.dart

```diff
@@ -140,57 +140,69 @@ environment:
         buildYaml,
         d.dir('lib', [d.file('builder.dart', '')]),
       ]).create();
-      await runPub('a', 'get');
+      var pubGetResult = await runPub('a', 'get');
+      expect(pubGetResult.exitCode, 0,
+          reason: 'stdout: ${pubGetResult.stdout}\n\n'
+              'stderr: ${pubGetResult.stderr}');
 
       final options = await findBuildScriptOptions(
           packageGraph: await PackageGraph.forPath('${d.sandbox}/a'));
       expect(options.canRunWithSoundNullSafety, isTrue);
     });
 
-    test('when the root package opts out', () async {
-      await d.dir('a', [
-        d.file('pubspec.yaml', '''
+    if (supportsUnsoundNullSafety) {
```

Comment:
We could (I would make this a group), but I don't think it matches the semantics of what we want here as much?

Generally I see a skipped test as something that should eventually be fixed (we show them with yellow warning text). These just don't make sense to run ever.

Reply:
Plus the `--run-skipped` flag would not make sense, these would fail.

Reply:
> Generally I see a skipped test as something that should eventually be fixed (we show them with yellow warning text).

The ones where we pass a String should eventually be fixed. I think the ones where we pass a bool it would be OK to consider as "valid" skips.

> Plus the `--run-skipped` flag would not make sense, these would fail (always).

That's a good point. I don't ever really run with that flag, but maybe I should every once in a while to catch stale skips.

Keeping as conditional blocks SGTM.